### PR TITLE
Enable rich LSIF hover information.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- This changeset adds a fake set of LSP client capabilities that the LSIF generator can pass into corresponding APIs. I've currently filled the capabilities with hover based information so that underlying systems can see that clients support Markdown and therefore.
- Updated tests to reflect the new behavior.

## Before
image

## After

image

### CODE REVIEW

**Review by dibarbet — APPROVED** [review]

**Review by mohaimenhasan — APPROVED** [review]
So cool !! Thank you for updating this !!

**Comment by NTaylorMullen** [comment]
Found I had broken some tests, all should be good now!

File: src/Features/Lsif/Generator/Generator.cs [hunk 1]
```diff
@@ -33,6 +34,20 @@ internal sealed class Generator
...
+                Hover = new LspProtocol.HoverSetting()
+                {
+                    ContentFormat = new[]
+                    {
+                        LspProtocol.MarkupKind.PlainText,
+                        LspProtocol.MarkupKind.Markdown,
+                    }
+                }
+            }
+        };
```
Comment (jasonmalinowski) [inline src/Features/Lsif/Generator/Generator.cs:50]: Blank line after this.

File: src/Features/Lsif/Generator/Generator.cs [hunk 2]
```diff
@@ -33,6 +34,20 @@ internal sealed class Generator
         private const bool FoldingRangeProvider = true;
         private const bool DiagnosticProvider = false;
 
+        private static readonly LspProtocol.ClientCapabilities LspClientCapabilities = new()
```
Comment (jasonmalinowski) [inline src/Features/Lsif/Generator/Generator.cs:37]: So here's a general question for LSIF: do we need a switch to the generator for things like this?

File: src/Features/Lsif/GeneratorTest/HoverTests.vb [hunk 3]
```diff
@@ -42,28 +42,46 @@ class C
             Dim expectedHoverContents As String
             Select Case code
                 Case "class [|C|] { string s; }"
-                    expectedHoverContents = "class C"
+                    expectedHoverContents = "```csharp
+class C
+```
+  "
```
Comment (jasonmalinowski) [inline src/Features/Lsif/GeneratorTest/HoverTests.vb:48]: What's with this magic two spaces here?

**Review by jasonmalinowski — APPROVED** [review]

File: src/Features/Lsif/Generator/Generator.cs (hunk 2 above)
Comment (jasonmalinowski) [reply src/Features/Lsif/Generator/Generator.cs:37]: (this question isn't blocking, but just something to think about)

Comment (NTaylorMullen) [reply src/Features/Lsif/Generator/Generator.cs:37]: Great question, I think eventually ya. IMO it shouldn't have too many knobs, maybe just  a "what LSP version do you holistically want me to generate for"?

File: src/Features/Lsif/GeneratorTest/HoverTests.vb (hunk 3 above)
Comment (jasonmalinowski) [reply src/Features/Lsif/GeneratorTest/HoverTests.vb:48]: Holy crap that's an, um, _interesting_ markdown behavior. I don't know why we're doing a single-spaced newline at the end of the contents though if there's nothing after it, but that's probably just a common issue with the hover thing so nothing unique here.

Comment (dibarbet) [reply src/Features/Lsif/GeneratorTest/HoverTests.vb:48]: indeed, see https://daringfireball.net/projects/markdown/syntax#p .  And you're right it probably shouldn't do that when the linebreak is the last 'tag'.

**Pull request merged**